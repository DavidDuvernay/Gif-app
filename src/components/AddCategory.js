import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { setCategories, categories } ) => {
 
    const [inputValue, setInputValue] = useState('');

    const handleinputChange = ( e ) => {
        
        setInputValue( e.target.value )
    };

    const handleSubmit =( e ) => {
        e.preventDefault();

        setCategories(categories => [ ...categories, inputValue]);
        setInputValue('');
    };

    return (
        <form onSubmit = { handleSubmit }>

            <input
                type = "text"
                value = { inputValue }
                onChange = { handleinputChange }
            /> 

        </form>
    )
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};