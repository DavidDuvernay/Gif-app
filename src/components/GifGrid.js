import React from 'react';

export const GifGrid = ( { category }) => {
    
    const getGifs = async() => {
        const url = 'http://api.giphy.com/v1/gifs/search?q=Rick%20and%20Morty&limit=10&api_key=DjBrCGeFD2LhaQNB1Lb05osHuPubFNcg';
        
        const resp = await fetch( url );
        
        const { data } = await resp.json();

        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        });

        console.log(gifs)
    };

    getGifs()

    return(
        <div> 
            <h3> { category } </h3>
        </div>
    );
    
}