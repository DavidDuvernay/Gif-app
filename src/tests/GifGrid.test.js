import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../components/GifGrid';
import { useFetchGifs } from '../hooks/useFetchGifs';
import '@testing-library/jest-dom';

jest.mock('../hooks/useFetchGifs');

describe('Pruebas en el componente', () => {
    const category= 'One Punch';

    test('Debe mostrar el componente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        
        const wrapper = shallow( <GifGrid category = { category } />);
        
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe de mostrar items cuando se cargan imagenes de useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/urldePrueba',
            title: 'Titulo de prueba',
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow( <GifGrid category = { category } />);

        expect( wrapper.find('p').exists()).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length )
    });
});