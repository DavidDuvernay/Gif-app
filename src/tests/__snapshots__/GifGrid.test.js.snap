// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Debe mostrar el componente 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__fadeInLeft"
  >
     
    One Punch
     
  </h3>
  <p
    className="card animate__animated animate__fadeInLeft"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
